### PR TITLE
custom collection sorting

### PR DESCRIPTION
**Pull request by ktarbet**
added code to sort collection similar to how DssVue sorts.

### CODE REVIEW

File: HEFSConverter/DssEnsembleReader.cs [hunk 1]
```diff
@@ -37,7 +37,31 @@ public IWatershedForecast Read(IHEFSReadArgs args)
...
+//            string previousT = dssPaths[0].Fpart.Split('|').Last().Split(':').Last();
+  //          string previousLoc = dssPaths[0].Bpart;
+            // /RUSSIANNAPA/APCC1/FLOW/01SEP2019/1HOUR/C:000002|T:0212019 
+
+            private string CollectionSortable(DSSPath x)
+            {
+                string rval = x.Apart + x.Bpart + x.Cpart + x.SortableDPart + x.Epart;
+                     
+                var tokens = x.Fpart.Split('|');
+                if (tokens.Length != 2)
```
Comment (HenryGeorgist) [inline HEFSConverter/DssEnsembleReader.cs:57]: what if there are more than 2 pipes? would we want to still alpha sort them?

File: HEFSConverter/DssEnsembleReader.cs [hunk 2]
```diff
@@ -37,7 +37,31 @@ public IWatershedForecast Read(IHEFSReadArgs args)
...
+                    return x.PathWithoutDate;
+                rval += tokens[1].Split(':')[1] + tokens[0].Split(':')[0];
```
Comment (HenryGeorgist) [inline HEFSConverter/DssEnsembleReader.cs:59]: wont tokens[0].Split(':')[0] always result in "C" - do you mean to put tokens[0].Split(':')[1]?

File: HEFSConverter/DssEnsembleReader.cs [hunk 3]
```diff
@@ -49,8 +73,12 @@ public ITimeSeriesOfEnsembleLocations ReadDataset(Watersheds watershed, DateTime
 
 			using (DSSIO.DSSReader dss = new DSSIO.DSSReader(dssPath))
 			{
-				DSSIO.DSSPathCollection dssPaths = dss.GetCondensedPathNames(); // sorted
-				if (dssPaths.Count == 0)
+				DSSIO.DSSPathCollection rawDssPaths = dss.GetCondensedPathNames(); // sorted
+                var dssPaths = rawDssPaths.OrderBy(a => a, new PathComparer()).ToArray(); // sorted
```
Comment (HenryGeorgist) [inline HEFSConverter/DssEnsembleReader.cs:77]: Does this mean we are sorting twice? what will that do to performance?

**Review by HenryGeorgist — APPROVED** [review]
overall it looks good, i am concerned about the tokens[0].Split(':')[0] -  i believe that is not your intent.

File: HEFSConverter/DssEnsembleReader.cs (hunk 2 above)
Comment (ktarbet) [reply HEFSConverter/DssEnsembleReader.cs:59]: Yes!  Thanks.

File: HEFSConverter/DssEnsembleReader.cs (hunk 3 above)
Comment (ktarbet) [reply HEFSConverter/DssEnsembleReader.cs:77]: Perhaps we can stop the timer, during sorting to eliminate this concern, during experimentation?

**Pull request merged**